assets/js/utils/filters.ts

```diff
@@ -50,7 +50,19 @@ export function changeUrl( newUrl: string ) {
 			''
 		);
 
-		// Reload the page with the updated URL
+		/**
+		 * If there are no query parameters in the URL and the URL ends with '?', we remove
+		 * the trailing '?' from the URL. The reason for this is that a '?' in the URL
+		 * typically indicates that query parameters will follow. If there are no query parameters,
+		 * the '?' is unnecessary and can cause the page to reload, which can negatively affect
+		 * performance. Therefore, by removing the '?', we prevent this unnecessary reload.
+		 */
+		const urlObject = new URL( newUrl );
+		const numberOfQueryParams = [ ...urlObject.searchParams.keys() ].length;
+		if ( numberOfQueryParams === 0 && newUrl.endsWith( '?' ) ) {
```

Comment:
I think it's safe to skip the query params check (`numberOfQueryParams === 0`), no? I mean, if the URL ends with `?`, it's always safe to remove the last question mark, no matter if there are query params or not, no? :thinking:

Reply:
I think so. I was just trying to be extra careful to ensure I do not break any existing functionality. 

But good point, the trailing "?" can indeed be removed safely without checking for query parameters. I will simplify the code accordingly. Thanks for the suggestion! 🙌🏻

Reply:
@Aljullu I have made the changes in [9a82cd9](https://github.com/woocommerce/woocommerce-blocks/pull/9952/commits/9a82cd95a27a1031d9e7b78475b334f812e1cf8b). Can you please take a look again just to be sure that everything still works as expected? 🙂